Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py

```diff
@@ -112,652 +711,840 @@ def remove_none_dict(input_dict: Dict) -> Dict:
-        'zerofox-modify-alert-tags': modify_alert_tags_command,
-        'zerofox-create-entity': create_entity_command,
-        'zerofox-list-alerts': list_alerts_command,
-        'zerofox-list-entities': list_entities_command,
-        'zerofox-get-entity-types': get_entity_types_command,
-        'zerofox-get-policy-types': get_policy_types_command,
-        'fetch-incidents': fetch_incidents,
+        "get-modified-remote-data": get_modified_remote_data_command,
+        "get-remote-data": get_remote_data_command,
+        "zerofox-get-alert": get_alert_command,
+        "zerofox-alert-user-assignment": alert_user_assignment_command,
+        "zerofox-close-alert": close_alert_command,
+        "zerofox-open-alert": open_alert_command,
+        "zerofox-alert-request-takedown": alert_request_takedown_command,
+        "zerofox-alert-cancel-takedown": alert_cancel_takedown_command,
+        "zerofox-modify-alert-tags": modify_alert_tags_command,
+        "zerofox-create-entity": create_entity_command,
+        "zerofox-list-alerts": list_alerts_command,
+        "zerofox-list-entities": list_entities_command,
+        "zerofox-get-entity-types": get_entity_types_command,
+        "zerofox-get-policy-types": get_policy_types_command,
+        "zerofox-modify-alert-notes": modify_alert_notes_command,
+        "zerofox-submit-threat": submit_threat_command,
+        "zerofox-search-compromised-domain": compromised_domain_command,
+        "zerofox-search-compromised-email": compromised_email_command,
+        "zerofox-search-malicious-ip": malicious_ip_command,
+        "zerofox-search-malicious-hash": malicious_hash_command,
+        "zerofox-search-exploits": search_exploits_command,
     }
     try:
+        client = ZFClient(
+            base_url=BASE_URL,
+            ok_codes={200, 201},
+            username=USERNAME,
+            password=PASSWORD,
+            fetch_limit=FETCH_LIMIT,
+        )
+
         handle_proxy()
-        commands[demisto.command()]()
+        command = demisto.command()
+
+        if command == 'test-module':
+            results = test_module(client)
+            return_results(results)
+        elif command == 'fetch-incidents':
+            next_run, incidents = fetch_incidents(
+                client,
+                last_run=demisto.getLastRun(),
+                first_fetch_time=FETCH_TIME,
+            )
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+        else:
+            command_handler = commands[command]
+            results = command_handler(client, demisto.args())
+            return_results(results)
 
     # Log exceptions
     except Exception as e:
         error_msg: str = str(e)
-        if demisto.command() == 'fetch-incidents':
+        if demisto.command() == "fetch-incidents":
             LOG(error_msg)
             LOG.print_log()
             raise
         else:
             return_error(error_msg)
```

Comment:
Since 'return_error' uses 'Log()' and 'Log.print_log()', no need to call them explicitly.
```suggestion
        return_error(str(e))
```